stream-chat-android-sample/src/main/kotlin/io/getstream/chat/sample/feature/login/LoginFragment.kt

```diff
@@ -6,63 +6,98 @@ import androidx.core.view.isVisible
 import androidx.fragment.app.Fragment
 import androidx.lifecycle.Observer
 import androidx.navigation.fragment.findNavController
+import com.getstream.sdk.chat.BuildConfig
 import io.getstream.chat.sample.R
 import io.getstream.chat.sample.application.EXTRA_CHANNEL_ID
 import io.getstream.chat.sample.common.navigateSafely
 import io.getstream.chat.sample.common.showToast
-import io.getstream.chat.sample.data.user.User
 import kotlinx.android.synthetic.main.fragment_login.*
+import kotlinx.android.synthetic.main.fragment_login.loadingProgressBar
 import org.koin.androidx.viewmodel.ext.android.viewModel
 
 class LoginFragment : Fragment(R.layout.fragment_login) {
 
     private val viewModel: LoginViewModel by viewModel()
 
-    private val adapter = UsersListAdapter {
-        viewModel.userClicked(this)
-    }
-
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
-        usersList.adapter = adapter
+        sdkVersion.text = getString(R.string.login_sdk_version_template, BuildConfig.VERSION_NAME)
+        developmentUsers.setOnClickListener {
+            redirectToUsersScreen()
+        }
+        loginButton.setOnClickListener {
+            viewModel.loginButtonClicked(collectCredentials())
+        }
+
         viewModel.state.observe(
             viewLifecycleOwner,
             Observer {
                 when (it) {
-                    is State.AvailableUsers -> renderAvailableUsers(it.availableUsers)
                     is State.RedirectToChannels -> redirectToChannelsScreen()
-                    is State.Loading -> changeLoadingIndicatorVisibility(true)
-                    is State.Error -> showErrorMessage(it.errorMessage)
                     is State.RedirectToChannel -> redirectToChannel(it.cid)
+                    is State.Loading -> showLoading()
+                    is State.Error -> showErrorMessage(it.errorMessage)
+                    is State.ValidationError -> showValidationErrors(it.invalidFields)
                 }
             }
         )
+
         activity?.intent?.getStringExtra(EXTRA_CHANNEL_ID)?.let { cid ->
             viewModel.targetChannelDataReceived(cid)
         }
     }
 
-    private fun redirectToChannel(cid: String) {
-        findNavController().navigateSafely(LoginFragmentDirections.actionLoginFragmentToChannelFragment(cid))
+    private fun collectCredentials(): LoginCredentials {
+        return LoginCredentials(
+            apiKey = apiKeyEditText.text.toString().trim(),
+            userId = userIdEditText.text.toString().trim(),
+            userToken = userTokenEditText.text.toString().trim(),
+            userName = userNameEditText.text.toString().trim()
```

Comment:
This could be cleaned up a bit with something like

```kotlin
private val EditText.trimmedText: String
        get() = text.trim().toString()
```